sdks/java/core/src/main/java/org/apache/beam/sdk/util/GcsUtil.java

```diff
@@ -483,12 +475,39 @@ public Void call() throws IOException {
       Thread.currentThread().interrupt();
       throw new IOException("Interrupted while executing batch GCS request", e);
     } catch (ExecutionException e) {
+      if (e.getCause() instanceof FileNotFoundException) {
```

Comment:
Is there a way to get all the exceptions from failed requests? Are they already contained in this one?